x-pack/plugins/transform/public/app/hooks/use_delete_transform.tsx

```diff
@@ -4,52 +4,273 @@
         if (results.hasOwnProperty(transformId)) {
-          if (results[transformId].success === true) {
-            toastNotifications.addSuccess(
-              i18n.translate('xpack.transform.transformList.deleteTransformSuccessMessage', {
-                defaultMessage: 'Request to delete transform {transformId} acknowledged.',
-                values: { transformId },
-              })
-            );
+          const status = results[transformId];
+          const destinationIndex = status.destinationIndex;
+
+          // if we are only deleting one modal, show the success toast messages
+          if (!isBulk) {
+            if (status.transformJobDeleted?.success) {
+              toastNotifications.addSuccess(
+                i18n.translate('xpack.transform.transformList.deleteTransformSuccessMessage', {
+                  defaultMessage: 'Request to delete transform job {transformId} acknowledged.',
+                  values: { transformId },
+                })
+              );
+            }
+            if (status.destIndexDeleted?.success) {
+              toastNotifications.addSuccess(
+                i18n.translate(
+                  'xpack.transform.deleteTransfrom.deleteAnalyticsWithIndexSuccessMessage',
+                  {
+                    defaultMessage:
+                      'Request to delete destination index {destinationIndex} acknowledged.',
+                    values: { destinationIndex },
+                  }
+                )
+              );
+            }
+            if (status.destIndexPatternDeleted?.success) {
+              toastNotifications.addSuccess(
+                i18n.translate(
+                  'xpack.transform.deleteTransfrom.deleteAnalyticsWithIndexPatternSuccessMessage',
+                  {
+                    defaultMessage:
+                      'Request to delete index pattern {destinationIndex} acknowledged.',
+                    values: { destinationIndex },
+                  }
+                )
+              );
+            }
           } else {
-            toastNotifications.addDanger(
-              i18n.translate('xpack.transform.transformList.deleteTransformErrorMessage', {
-                defaultMessage: 'An error occurred deleting the transform {transformId}',
+            Object.keys(successCount).forEach((key) => {
+              // @ts-ignore
+              if (status[key]?.success) {
+                successCount[key] = successCount[key] + 1;
+              }
+            });
+          }
+          if (status.transformJobDeleted?.error) {
+            const error = extractErrorMessage(status.transformJobDeleted.error);
+            toastNotifications.addDanger({
+              title: i18n.translate('xpack.transform.transformList.deleteTransformErrorMessage', {
+                defaultMessage:
+                  'An error occurred deleting the data frame analytics job {transformId}',
```

Comment:
Copy paste error :)  Should be `An error occurred deleting the transform {transformId}`.